### PR TITLE
Replace all uses of core::list with std::list

### PR DESCRIPTION
**Pull request by paradust7**
Part of https://github.com/minetest/minetest/issues/12312

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]

File: src/gui/mainmenumanager.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class MainMenuManager : public IMenuManager
 		// Remove all entries if there are duplicates
 		m_stack.remove(menu);
 
-		/*core::list<GUIModalMenu*>::Iterator i = m_stack.getLast();
```
Comment (nerzhul) [inline src/gui/mainmenumanager.h:67]: just remove this commented part

**Review by nerzhul — CHANGES_REQUESTED** [review]
please fix build too

File: src/gui/mainmenumanager.h (hunk 1 above)
Comment (paradust7) [reply src/gui/mainmenumanager.h:67]: done

**Comment by paradust7** [comment]
The build is broken because it depends on the corresponding irrlicht change: https://github.com/minetest/irrlicht/pull/105

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**